examples/esp_netif/slip_custom_netif/main/slip_client_main.c

```diff
@@ -0,0 +1,236 @@
+             lwip_ntohl(addr.addr[0]), lwip_ntohl(addr.addr[1]));
+
+    // Build slip set message
+    buff[0] = '!';
+    buff[1] = 'P';
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 4; j++) {
+            buff[2 + i * 4 + j] = addr.addr[i] >> (j * 8);
+        }
+    }
+
+    // Write raw data out the slip interface
+    slip_modem_raw_write(slip, buff, 2 + 8);
+}
+
+// slip_rx_filter filters incoming commands from the slip interface
+// this implementation is designed for use with contiki slip devices
+static bool slip_rx_filter(slip_modem_t *slip, uint8_t *data, uint32_t len)
+{
+    if (data[1] == '?') {
+        switch (data[2]) {
+        case 'P':
+            ESP_LOGI(TAG, "Prefix request");
+            slip_set_prefix(slip);
+
+            return true;
+
+        default:
+            ESP_LOGI(TAG, "Unhandled request '%c'", data[2]);
+            break;
+        }
+
+        return true;
+
+    } else if (data[1] == '!') {
+        switch (data[2]) {
+        default:
+            ESP_LOGI(TAG, "Unhandled command '%c'", data[2]);
+            break;
+        }
+    }
+
+    return false;
+}
+
+#if CONFIG_EXAMPLE_IPV4
+static const esp_netif_ip_info_t s_slip_ip4 = {
+        .ip = { .addr = ESP_IP4TOADDR( 10, 0, 0, 2) },
+};
+#endif
+
+// Initialise the SLIP interface
+esp_netif_t *slip_if_init(void)
+{
+    ESP_LOGI(TAG, "Initialising SLIP interface");
+
+    esp_netif_inherent_config_t base_cfg = ESP_NETIF_INHERENT_DEFAULT_SLIP()
+#if CONFIG_EXAMPLE_IPV4
+    base_cfg.ip_info = &s_slip_ip4;
+#endif
+    esp_netif_config_t cfg = {          .base = &base_cfg,
+                                        .driver = NULL,
+                                        .stack = netstack_default_slip };
+
+    esp_netif_t *slip_netif = esp_netif_new(&cfg);
+
+    esp_ip6_addr_t local_addr;        /* Local IP6 address */
+    IP6_ADDR(&local_addr,
+             lwip_htonl(0xfd0000),
+             lwip_htonl(0x00000000),
+             lwip_htonl(0x00000000),
+             lwip_htonl(0x00000001)
+            );
+
+    ESP_LOGI(TAG, "Initialising SLIP modem");
+
+    slip_modem_config_t modem_cfg = {
+        .uart_dev = UART_NUM_1,
+
+        .uart_tx_pin = CONFIG_EXAMPLE_UART_TX_PIN,
+        .uart_rx_pin = CONFIG_EXAMPLE_UART_RX_PIN,
+        .uart_baud = CONFIG_EXAMPLE_UART_BAUD,
+
+        .rx_buffer_len = 1024,
+        .rx_filter = slip_rx_filter,
+        .ipv6_addr = &local_addr
+    };
+
+    void *slip_modem = slip_modem_create(slip_netif, &modem_cfg);
+    assert(slip_modem);
+    ESP_ERROR_CHECK(esp_netif_attach(slip_netif, slip_modem));
+
+    ESP_LOGI(TAG, "SLIP init complete");
+
+    return slip_netif;
+}
+
+void app_main(void)
+{
+    // Setup networking
+    esp_netif_init();
+
+    esp_log_level_set("*", ESP_LOG_DEBUG);
+
+    // Create event loop
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+
+    // Setup slip interface
+    slip_if_init();
+
+    // Setup UDP loopback service
+    udp_rx_tx_init();
```

Comment:
It would be nice to have an explanation on what the user achieved here and how to use the created interface. E.g. why is it ok to ignore the return from `slip_if_init`?

Reply:
Good catch, thanks! Have added a null check to the return value of `slip_if_init()`.

As for the other API, this is the user defined function (operates on UDP sockets), have updated the comment and made the function `static`.